app/services/vacation_work_times_service.rb

```diff
@@ -4,15 +4,18 @@ class VacationWorkTimesService
   def initialize(vacation, current_user)
     @vacation = vacation
     @current_user = current_user
-    @project = Project.where(name: 'Vacation').first
+    @project = @vacation.illness? ? Project.find_by(name: 'ZKS') : Project.find_by(name: 'Vacation')
     @user = User.find(@vacation.user_id)
     @vacation_range = @vacation.start_date.business_dates_until(@vacation.end_date + 1.day)
   end
 
   def save
-    return if !@current_user.staff_manager? || work_times.any?
+    return unless @current_user.staff_manager?
 
+    work_time_dates = work_times.pluck(:starts_at).map { |date| date.strftime('%Y-%m-%d') }
```

Comment:
looks like we could just convert starts_at to date instead of formatting. So i.e.
work_time_dates = work_times.pluck(:starts_at).map(&:to_date)
next if work_time_dates.include?(day.to_date)